File: shoring/flywheel/src/test/java/com/panopset/tests/flywheel/SimpleTest.java
package com.panopset.tests.flywheel;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.io.File;
import java.io.IOException;

import org.junit.jupiter.api.Test;

import com.panopset.compat.Fileop;
import com.panopset.flywheel.Flywheel;
import com.panopset.flywheel.FlywheelBuilder;
import com.panopset.gp.FileCompare;
import com.panopset.tests.TestDirectory;

public class SimpleTest implements TestDirectory {

	// TODO
	public static final String TEST_FILE_PATH = "src/test/resources/com/panopset/tests/flywheel/";


	public static final String SIMPLETEST = "simpleTest.txt";
	public static final String SIMPLEOUT = "simpleOut.txt";
	public static final String EXPECTED = "simpleTestExpected.txt";

	@Test
	void testSimpleScript() throws IOException {
		new SimpleTest().comparisonTest(SIMPLETEST, SIMPLEOUT, EXPECTED);
	}

	public void comparisonTest(final String scriptName, final String generatedFileName, final String expected)
			throws IOException {
		comparisonTest(scriptName, new String[] { generatedFileName }, new String[] { expected });
	}

	public void comparisonTest(final String scriptName, final String[] generatedFileNames, final String[] expecteds)
			throws IOException {
		assertEquals(expecteds.length, generatedFileNames.length);
		int incr = 0;
		for (String generatedFileName : generatedFileNames) {
			File generatedFile = new File(Fileop.combinePaths(TEST_DIRECTORY, generatedFileName));
			if (generatedFile.exists()) {
				Fileop.delete(generatedFile);
			}
			Flywheel script = new FlywheelBuilder().targetDirectory(TEST_DIRECTORY)
					.file(new File(String.format("%s%s", TEST_FILE_PATH, scriptName))).construct();
			script.exec();
			assertFalse(script.isStopped());
			assertTrue(generatedFile.exists());
			assertTrue(FileCompare.filesAreSame(generatedFile,
					new File(String.format("%s%s", TEST_FILE_PATH, expecteds[incr++]))));
		}
	}

}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/flywheel/SimpleTest.kt
package com.panopset.tests.flywheel

import com.panopset.compat.Fileop
import com.panopset.flywheel.FlywheelBuilder
import com.panopset.gp.FileCompare
import com.panopset.tests.TEST_DIRECTORY
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File

class SimpleTest {
    @Test
    fun testSimpleScript() {
        SimpleTest().comparisonTest(SIMPLETEST, SIMPLEOUT, EXPECTED)
    }

    fun comparisonTest(scriptName: String?, generatedFileName: String, expected: String) {
        comparisonTest(scriptName, arrayOf(generatedFileName), arrayOf(expected))
    }

    fun comparisonTest(scriptName: String?, generatedFileNames: Array<String>, expecteds: Array<String>) {
        Assertions.assertEquals(expecteds.size, generatedFileNames.size)
        for ((incr, generatedFileName) in generatedFileNames.withIndex()) {
            val generatedFile = File(Fileop.combinePaths(TEST_DIRECTORY, generatedFileName))
            if (generatedFile.exists()) {
                Fileop.delete(generatedFile)
            }
            val script = FlywheelBuilder().targetDirectory(TEST_DIRECTORY)
                .file(File(String.format("%s%s", TEST_FILE_PATH, scriptName))).construct()
            script.exec()
            Assertions.assertFalse(script.isStopped)
            Assertions.assertTrue(generatedFile.exists())
            Assertions.assertTrue(
                FileCompare.filesAreSame(
                    generatedFile,
                    File(String.format("%s%s", TEST_FILE_PATH, expecteds[incr]))
                )
            )
        }
    }

    companion object {
        // TODO
        const val TEST_FILE_PATH = "src/test/resources/com/panopset/tests/flywheel/"
        const val SIMPLETEST = "simpleTest.txt"
        const val SIMPLEOUT = "simpleOut.txt"
        const val EXPECTED = "simpleTestExpected.txt"
    }
}
